HerPublicWebsite.UnitTests/BusinessLogic/ExternalServices/EmailSending/GovUkNotifyApiTests.cs

```diff
@@ -42,19 +47,31 @@ public void SendPendingReferralReportEmail_WhenCalled_CallsSendEmailOnEmailInCon
         config.PendingReferralEmailRecipients = recipient;
         
         // Act
-        govUkNotifyApi.SendPendingReferralReportEmail();
+        govUkNotifyApi.SendPendingReferralReportEmail(memoryStream);
         
         // Assert
         mockNotificationClient.Verify(nc => nc.SendEmail(
             recipient,
             config.PendingReferralReportTemplate.Id,
-            new Dictionary<string, object>
-            {
-                { config.PendingReferralReportTemplate.LinkPlaceholder, "https://www.gov.uk/apply-home-upgrade-grant" }
-            },
+            It.IsAny<Dictionary<string, object>>(),
             null, null));
     }
 
+    [Test]
+    public void SendPendingReferralReportEmail_WhenCalled_IncludesALinkInPersonalisation()
+    {
+        // Arrange
+        const string recipient = "email1@example.com";
+        config.PendingReferralEmailRecipients = recipient;
+        
+        // Act
+        govUkNotifyApi.SendPendingReferralReportEmail(memoryStream);
+        
+        // Assert
+        var personalisation = (Dictionary<string, object>)mockNotificationClient.Invocations[0].Arguments[2];
+        personalisation.Should().ContainKey("link");
```

Comment:
we did consider this, though we weren't sure if there was anything useful we could verify about the value for key "link", since it seems the file upload itself happens later

Reply:
Hmm, OK. I guess we'll have to leave this for now. I think it would be possible to interrogate the `JObject` produced to check that it matches what we expect, but the testing we have already gives us quite good coverage.